keanu-project/src/test/java/io/improbable/keanu/network/BayesianNetworkTest.java

```diff
@@ -102,4 +107,68 @@ public void networkWithNonSaveableVerticesThrowsExceptionOnSave() throws IOExcep
         NetworkSaver netSaver = mock(NetworkSaver.class);
         net.save(netSaver);
     }
+
+    @Test
+    public void youCanGetVertexCountOfInitialBayesNet() {
+        ConstantDoubleVertex probTrue = ConstantVertex.of(0.5);
+        BoolVertex a = new BernoulliVertex(probTrue);
+        Set<Vertex> connectedGraph = a.getConnectedGraph();
+        BayesianNetwork net = new BayesianNetwork(connectedGraph);
+
+        BoolVertex b = new BernoulliVertex(0.5);
```

Comment:
This seems to be testing that adding vertices after a Bayes net has been created doesn't change the vertex count. I think this is worthy of it's own test, would be much clearer as to what's going on and the expected behaviour.